From 7ba0a952949410eeb2b0ae81427d31697bbd7886 Mon Sep 17 00:00:00 2001
From: Diego Escalante Urrelo <descalante@igalia.com>
Date: Wed, 13 Jan 2010 23:43:58 +0000
Subject: Use g_mkdir_with_parents instead of just mkdir

This ensures that if .gnome2/ doesn't exist, we don't fail instead of just
creating the dir.

Bug #605860
---
diff --git a/lib/ephy-file-helpers.c b/lib/ephy-file-helpers.c
index 8777c96..271d362 100644
--- a/lib/ephy-file-helpers.c
+++ b/lib/ephy-file-helpers.c
@@ -372,7 +372,7 @@ ephy_ensure_dir_exists (const char *dir,
 	}
 
 	if (!g_file_test (dir, G_FILE_TEST_EXISTS) &&
-            mkdir (dir, 488) != 0)
+            g_mkdir_with_parents (dir, 488) != 0)
 	{
 		g_set_error (error,
 			     EPHY_FILE_HELPERS_ERROR_QUARK,
--
cgit v0.8.3.1
